WordPress/src/main/java/org/wordpress/android/ui/FilteredRecyclerView.java

```diff
@@ -526,6 +608,89 @@ public boolean isFirstItemVisible() {
         return (child != null && mRecyclerView.getLayoutManager().getPosition(child) == 0);
     }
 
+    class FilteredPagerAdapter extends PagerAdapter {
+        private RecyclerView mRecyclerView;
+        private List<FilterCriteria> mFilterCriteria;
+
+        FilteredPagerAdapter(RecyclerView recyclerView, List<FilterCriteria> filterCriteria) {
+            mRecyclerView = recyclerView;
+            mFilterCriteria = filterCriteria;
+        }
+
+
+        private boolean datasetChanged(List<FilterCriteria> filterCriteriaOptions) {
+            boolean isChanged = false;
+
+            if (mFilterCriteria.size() != filterCriteriaOptions.size()) {
+                isChanged = true;
+            } else {
+                for (int index = 0; index < mFilterCriteria.size(); index++) {
+                    if (!mFilterCriteria.get(index).equals(filterCriteriaOptions.get(index))) {
+                        isChanged = true;
+                        break;
+                    }
+                }
+            }
+
+            return isChanged;
+        }
+
+        public void updateFiltersIfNeeded(List<FilterCriteria> filterCriteriaOptions) {
+            if (datasetChanged(filterCriteriaOptions)) {
+                mFilterCriteria = filterCriteriaOptions;
+                mSelectingRememberedFilterOnCreate = true;
+                notifyDataSetChanged();
+            }
+        }
+
+        public FilterCriteria getFilterAtPosition(int position) {
+            return mFilterCriteria.get(position);
+        }
+
+        @Override public int getItemPosition(@NonNull Object object) {
+            if (object != null && mFilterCriteria != null) {
```

Comment:
Do you need the `!= null` check if the parameter is marked as `NonNull`? :-)

Reply:
well...no...probably 😊...changed here fedddd9c0601c992dec63becacd01b68dd51f11b